conn/physic/units_test.go

```diff
@@ -857,6 +1019,166 @@ func TestValueOfUnitString(t *testing.T) {
 	}
 }
 
+func TestAngleSet(t *testing.T) {
+	succeeds := []struct {
+		in       string
+		expected Angle
+	}{
+		{"1nrad", 1 * NanoRadian},
+		{"10nrad", 10 * NanoRadian},
+		{"100nrad", 100 * NanoRadian},
+		{"1urad", 1 * MicroRadian},
+		{"10urad", 10 * MicroRadian},
+		{"100urad", 100 * MicroRadian},
+		{"1µrad", 1 * MicroRadian},
+		{"10µrad", 10 * MicroRadian},
+		{"100µrad", 100 * MicroRadian},
+		{"1mrad", 1 * MilliRadian},
+		{"10mrad", 10 * MilliRadian},
+		{"100mrad", 100 * MilliRadian},
+		{"1rad", 1 * Radian},
+		{"10rad", 10 * Radian},
+		{"100rad", 100 * Radian},
+		{"1krad", 1000 * Radian},
+		{"10krad", 10000 * Radian},
+		{"100krad", 100000 * Radian},
+		{"1Mrad", 1000000 * Radian},
+		{"10Mrad", 10000000 * Radian},
+		{"100Mrad", 100000000 * Radian},
+		{"1Grad", 1000000000 * Radian},
+		{"12.345rad", 12345 * MilliRadian},
+		{"-12.345rad", -12345 * MilliRadian},
+		{"9.223372036854775807Grad", 9223372036854775807 * NanoRadian},
+		{"-9.223372036854775807Grad", -9223372036854775807 * NanoRadian},
+		{"1deg", 1 * Degree},
+		{"1Mdeg", 1000000 * Degree},
+		{"100Gdeg", 100000000000 * Degree},
+		{"500Gdeg", 500000000000 * Degree},
+		{"528460276055deg", 528460276055 * Degree},
+		{"1mdeg", Degree / 1000},
+		{"1udeg", Degree / 1000000},
+	}
+
+	fails := []struct {
+		in  string
+		err string
+	}{
+		{
+			"10000000000Tdeg",
+			"exponent exceeds int64",
+		},
+		{
+			"10Trad",
+			"exponent exceeds int64",
+		},
+		{
+			"10Erad",
+			"contains unknown unit prefix \"E\". valid prefixes for \"Rad\" are p,n,u,µ,m,k,M,G or T",
+		},
+		{
+			"10Exarad",
+			"contains unknown unit prefix \"Exa\". valid prefixes for \"Rad\" are p,n,u,µ,m,k,M,G or T",
+		},
+		{
+			"10eRadianE",
+			"contains unknown unit prefix \"e\". valid prefixes for \"Rad\" are p,n,u,µ,m,k,M,G or T",
+		},
+		{
+			"10",
+			"no units provided, need Rad",
+		},
+		{
+			"9223372036854775808rad",
```

Comment:
fmt.Sprintf("%drad", maxAngle + 1) ?

Reply:
could do `fmt.Sprintf("%drad", uint64(maxAngle)+1)`?

Reply:
In tests it's fine to use fmt.